building_sim_plugins/building_sim_gazebo/src/door.cpp

```diff
@@ -5,217 +5,88 @@
-
-  double _last_update_time;
-  double _last_pub_time;
+  std::shared_ptr<DoorCommon> _door_common = nullptr;
 
   bool _initialized = false;
 
 public:
   DoorPlugin()
-  : _last_update_time(0),
-    _last_pub_time(0)
   {
     // Do nothing
   }
 
   void Load(gazebo::physics::ModelPtr model, sdf::ElementPtr sdf) override
   {
-    _ros_node = gazebo_ros::Node::Get(sdf);
+    auto _ros_node = gazebo_ros::Node::Get(sdf);
     _model = model;
 
     RCLCPP_INFO(
       _ros_node->get_logger(),
       "Loading DoorPlugin for [%s]",
       _model->GetName().c_str());
 
-    MotionParams params;
-    get_sdf_param_if_available<double>(sdf, "v_max_door", params.v_max);
-    get_sdf_param_if_available<double>(sdf, "a_max_door", params.a_max);
-    get_sdf_param_if_available<double>(sdf, "a_nom_door", params.a_nom);
-    get_sdf_param_if_available<double>(sdf, "dx_min_door", params.dx_min);
-    get_sdf_param_if_available<double>(sdf, "f_max_door", params.f_max);
+    _door_common = DoorCommon::make(
+      _model->GetName(),
+      _ros_node,
+      sdf);
 
-    sdf::ElementPtr door_element;
-    std::string left_door_joint_name;
-    std::string right_door_joint_name;
-    std::string door_type;
-    if (!get_element_required(sdf, "door", door_element) ||
-      !get_sdf_attribute_required<std::string>(
-        door_element, "left_joint_name", left_door_joint_name) ||
-      !get_sdf_attribute_required<std::string>(
-        door_element, "right_joint_name", right_door_joint_name) ||
-      !get_sdf_attribute_required<std::string>(
-        door_element, "type", door_type))
-    {
-      RCLCPP_ERROR(
-        _ros_node->get_logger(),
-        " -- Missing required parameters for [%s] plugin",
-        _model->GetName().c_str());
+    if (!_door_common)
       return;
-    }
 
-    if (left_door_joint_name == "empty_joint" &&
-      right_door_joint_name == "empty_joint")
-    {
-      RCLCPP_ERROR(
-        _ros_node->get_logger(),
-        " -- Both door joint names are missing for [%s] plugin, at least one"
-        " is required", _model->GetName().c_str());
-      return;
-    }
+    const auto left_door_joint_name = _door_common->left_door_joint_name();
+    const auto right_door_joint_name = _door_common->right_door_joint_name();
 
     if (left_door_joint_name != "empty_joint")
     {
-      const auto left_door_joint = _model->GetJoint(left_door_joint_name);
-      if (!left_door_joint)
+      _left_door_joint = _model->GetJoint(left_door_joint_name);
+      if (!_left_door_joint)
       {
         RCLCPP_ERROR(
           _ros_node->get_logger(),
           " -- Model is missing the left door joint [%s]",
           left_door_joint_name.c_str());
         return;
       }
-      _doors.emplace_back(_model->GetName() == "chart_lift_door",
-        left_door_joint, params);
+      _door_common->add_left_door(_model->GetName() == "chart_lift_door",
+        _left_door_joint->UpperLimit(0), _left_door_joint->LowerLimit(0));
```

Comment:
The joint limits are now inferred from the parent sdf in `make()`. However with the ign version, `sdf::ElementPtr::GetParent()` method always returned a `nullptr`.I think this might be a bug. As a workaround, I'm using default values of the limits when this method fails. 
https://github.com/osrf/traffic_editor/blob/7a7984bd422248222fb26c69181da8b872ca4906/building_sim_plugins/building_sim_common/include/building_sim_common/door_common.hpp#L202